otelcolbuilder/cmd/configprovider.go

```diff
@@ -63,53 +56,23 @@ func UseCustomConfigProvider(params *otelcol.CollectorSettings) error {
 	}
 
 	locations := getConfigFlag(flagset)
-	opAmpPath := flagset.Lookup(opAmpConfigFlag)
+	opAmpPath := getOpampConfigFlag(flagset)
 
-	if len(locations) > 0 && opAmpPath.Value.String() != "" {
+	if len(locations) > 0 && opAmpPath != "" {
 		return fmt.Errorf("cannot use --%s and --%s flags together", configFlag, opAmpConfigFlag)
 	}
 
-	if len(locations) == 0 && opAmpPath.Value.String() == "" {
-		// if no locations, use defaults
-		// either this is a command, or the default provider will throw an error
+	if opAmpPath == "" {
+		// let the default config provider handle things, we don't need to do anything
 		return nil
 	}
 
-	// create the config provider using the locations
-	if len(locations) > 0 {
-		params.ConfigProviderSettings = NewConfigProviderSettings(locations)
```

Comment:
don't we need this to support other configproviders if the opamp path isn't set? or is the builder ensuring this is done?

Reply:
This was actually already doable before the builder change, we just didn't take advantage of it. The only reason this code was here was to add config locations to the settings, and that's not necessary anymore since https://github.com/open-telemetry/opentelemetry-collector/pull/9228.

Reply:
But yes, now we *also* don't need to specify the providers here, they're specified in the `main.go` generated by the builder.